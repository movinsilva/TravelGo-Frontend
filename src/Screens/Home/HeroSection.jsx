import { Container, Button } from 'react-bootstrap';
import './HeroSection.scss';
import { useGetStationsQuery } from '../../slices/trainApiSlice';
import { LinkContainer } from 'react-router-bootstrap';


const HeroSection = () => {
  const {data, isLoading} = useGetStationsQuery();


  console.log("ðŸš€ ~ file: HeroSection.jsx:8 ~ HeroSection ~ data:", data)


  return (
    <section className="hero-section">
      <div className="hero-background"></div>

      <Container className="hero-content">
      <div><p>Hello Travellers</p>
        <h1>make your booking<br />
        experience easy!</h1></div>
      
        <div className="search-form">
          <div className="glass-container-extend">
            <select>
              {(data!=undefined) ? data.map((station) => (
                <option>{station.StationName}</option>
              )) : <><option>Loading</option></>}
           
            </select>
            <input type="text" placeholder="From" />
            <input type="text" placeholder="To" />
            <input type="date" placeholder="Date" />
            
            <LinkContainer to="/booking">
            <Button variant="primary" className='button-extend'>Search</Button>
            </LinkContainer>
          </div>
        </div>
      </Container>
    </section>
  );
};

export default HeroSection;
