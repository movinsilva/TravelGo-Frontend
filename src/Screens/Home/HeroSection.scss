@import '../Common/main.scss';

.hero-section {
    position: relative;
    height: 100vh;
    overflow: hidden;
  
    .hero-background {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-image: url('../../assets/Background\ Img.png');
      background-size: cover;
      z-index: -1;
    }

    .glass-container-extend{
      @include glass-container;
        padding: 30px; 
        display: flex;
    }
    .hero-content {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: left;
      height: 100%;
      margin-top: 2%;
  
      h1 {
        font-size: 3rem;
        color: #fff;
        text-align: left;
        margin-bottom: 10px;
        font-weight: weight("medium-l");
        letter-spacing: 0.3px;
      }
      p {
        font-size: 1.2rem;
        color: #fff;
        text-align: center;
        background-color: $primary-color;
        border-radius: 35px;
        padding: 5px 18px 5px 18px;
        width: fit-content;
        margin-bottom: 0.8%;
        
      }
  
      .search-form {
        display: flex;
        align-items: center;
        margin: 10% auto 0 auto;
        input[type='text'],
        input[type='date'] {
          padding: 13px;
          margin-right: 10px;
          border-radius: 5px;
          background: none;
          border: 2px solid rgba(255, 255, 255, 0.715);
     
        }

        input::placeholder {
          color: #000000; /* Red color for the placeholder text */
        }

  
        .button-extend {
          @include Button;
        }
      }

      
    }
  }
 
@media(max-width:920px){
  .hero-section{

    .hero-content{
      align-items: center;
      margin-top: -1%;
      justify-content: center;
      h1{
        display: none;
      }
      
      .search-form{
        
        .glass-container-extend{
          display: flex;
          flex-direction: column;
          padding: 6%;
          width: 100%;
        }
        input[type='text'], select {
          width: 100%;
        }
        input[type='date'] { 
          width: 100%;
        }
        .button-extend{
          width: 100%;
        }
      }
    }
  }
}  
  