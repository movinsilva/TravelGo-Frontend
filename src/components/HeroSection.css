.hero-section {
  position: relative;
  height: 100vh;
  overflow: hidden;
}
.hero-section .hero-background {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-image: url("../assets/Background Img.png");
  background-size: cover;
  filter: brightness(0.6);
  z-index: -1;
}
.hero-section .hero-content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
}
.hero-section .hero-content h1 {
  font-size: 3rem;
  color: #fff;
  text-align: center;
  margin-bottom: 10px;
}
.hero-section .hero-content p {
  font-size: 1.5rem;
  color: #fff;
  text-align: center;
  margin-bottom: 30px;
}
.hero-section .hero-content .search-form {
  display: flex;
  align-items: center;
}
.hero-section .hero-content .search-form .glass-container {
  background: rgba(255, 255, 255, 0.2);
  border-radius: 10px;
  padding: 20px;
  display: flex;
  flex-direction: row;
  gap: 10px;
  -webkit-backdrop-filter: blur(10px);
          backdrop-filter: blur(10px);
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}
.hero-section .hero-content .search-form input[type=text],
.hero-section .hero-content .search-form input[type=date] {
  padding: 10px;
  margin-right: 10px;
  border: 1px solid rgba(255, 255, 255, 0.2);
  border-radius: 5px;
  background: none;
}
.hero-section .hero-content .search-form button {
  padding: 10px 20px;
  border-radius: 5px;
}/*# sourceMappingURL=HeroSection.css.map */