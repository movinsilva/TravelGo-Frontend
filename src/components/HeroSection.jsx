import React from 'react';
import { Container, Button } from 'react-bootstrap';
import './HeroSection.scss';

const HeroSection = () => {
  return (
    <section className="hero-section">
      <div className="hero-background"></div>
      <Container className="hero-content">
        <h1>Welcome to Train Booking System</h1>
        <p>Your journey starts here</p>
        <div className="search-form">
          <div className="glass-container">
            <input type="text" placeholder="From" />
            <input type="text" placeholder="To" />
            <input type="date" placeholder="Date" />
            <Button variant="primary">Search</Button>
          </div>
        </div>
      </Container>
    </section>
  );
};

export default HeroSection;
