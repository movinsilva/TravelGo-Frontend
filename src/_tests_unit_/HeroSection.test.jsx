import { describe, expect, it} from "vitest";
import {render,screen,fireEvent} from '@testing-library/react';
import TestRenderer from "react-test-renderer"; 
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux"; // Import the Provider
import store from "../store.js"; // Import your Redux store
import HeroSection from './../Screens/Home/HeroSection';
import "@testing-library/jest-dom";


describe("Home Page/HeroSection.jsx",()=>{
    it("Input fields takes correct input types",()=>{
        const tree= TestRenderer.create(
        <Provider store={store}>
            <MemoryRouter>
                <HeroSection/>
            </MemoryRouter>
        </Provider>).toJSON();

        expect(tree.children[1].children[1].children[0].children[0].props.className).toBe("dropdown-class")
        expect(tree.children[1].children[1].children[0].children[1].props.className).toBe("dropdown-class")
        expect(tree.children[1].children[1].children[0].children[2].props.className).toBe("dropdown-class")

        expect(tree.children[1].children[1].children[0].children[2].children[1].props.type).toBe("date")
        //console.log(tree.children[1].children[1].children[0].children[2].children[1])



        expect(tree).toMatchSnapshot(); // This can be used for snapshot testing
    })

    it("Web page elements Renders correctly",()=>{
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <HeroSection/>
                </MemoryRouter>
            </Provider>
        )
        expect(screen.getByText("Hello Travellers")).toBeInTheDocument();
        expect(screen.getByText("From")).toBeInTheDocument();
        expect(screen.getByText("To")).toBeInTheDocument();
        expect(screen.getByText("Date")).toBeInTheDocument();
        expect(screen.getByText("Search")).toBeInTheDocument();


    })
    

   
})