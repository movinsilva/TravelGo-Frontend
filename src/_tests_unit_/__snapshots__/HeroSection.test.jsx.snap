// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page/HeroSection.jsx > Input fields takes correct input types 1`] = `
<section
  className="hero-section"
>
  <div
    className="hero-background"
  />
  <div
    className="hero-content container"
  >
    <div>
      <p>
        Hello Travellers
      </p>
      <h1>
        make your booking
        <br />
        experience easy!
      </h1>
    </div>
    <div
      className="search-form"
    >
      <div
        className="glass-container-extend"
      >
        <div
          className="dropdown-class"
        >
          <label
            className="dropdown-label"
          >
            From
          </label>
          <select>
            <option>
              Loading
            </option>
          </select>
        </div>
        <div
          className="dropdown-class"
        >
          <label
            className="dropdown-label"
          >
            To
          </label>
          <select>
            <option>
              Loading
            </option>
          </select>
        </div>
        <div
          className="dropdown-class"
        >
          <label
            className="dropdown-label"
          >
            Date
          </label>
          <input
            placeholder="Date"
            type="date"
          />
        </div>
        <div
          className="search-btn"
        >
          <a
            className="button-extend btn btn-primary"
            disabled={false}
            href="/booking"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={null}
            tabIndex={0}
          >
            Search
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Home Page/HeroSection.jsx > Renders correctly 1`] = `
<section
  className="hero-section"
>
  <div
    className="hero-background"
  />
  <div
    className="hero-content container"
  >
    <div>
      <p>
        Hello Travellers
      </p>
      <h1>
        make your booking
        <br />
        experience easy!
      </h1>
    </div>
    <div
      className="search-form"
    >
      <div
        className="glass-container-extend"
      >
        <div
          className="dropdown-class"
        >
          <label
            className="dropdown-label"
          >
            From
          </label>
          <select>
            <option>
              Loading
            </option>
          </select>
        </div>
        <div
          className="dropdown-class"
        >
          <label
            className="dropdown-label"
          >
            To
          </label>
          <select>
            <option>
              Loading
            </option>
          </select>
        </div>
        <div
          className="dropdown-class"
        >
          <label
            className="dropdown-label"
          >
            Date
          </label>
          <input
            placeholder="Date"
            type="date"
          />
        </div>
        <div
          className="search-btn"
        >
          <a
            className="button-extend btn btn-primary"
            disabled={false}
            href="/booking"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={null}
            tabIndex={0}
          >
            Search
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
